alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/jfx/events/generic/EventDispatcher.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.jfx.events.generic
+
+import javafx.event.Event
+
+/**
+ * An event dispatcher.
+ * @param T the type of the action this dispatcher models
+ * @param E the type of event that triggers this dispatcher
+ */
+interface EventDispatcher<T : TriggerAction, E : Event> {
```

Comment:
contravariant?

Reply:
`T` is contravariant, but `E` is invariant because it is used contravariantly for `val listener` but covariantly for `fun set` since `(E) -> Unit` is equal to `Consumer<E>`.